p2p/config/config.go

```diff
@@ -20,6 +20,8 @@ const (
 	defaultTCPPort = 7513
 	// defaultTCPInterface is the inet interface that P2P listens on by default
 	defaultTCPInterface = "0.0.0.0"
+	// max number of parallel gossip goroutines per channel (protocol)
+	defaultMaxGossipRoutines = 50
```

Comment:
a default needs to be based on what the task is doing. if this is a cpu task the number of goroutines shouldn't be higher than runtime.NumCPU, if writing to the database it probably should be runtime.NumCPU as well since we are not flushing to the disk currently. otherwise it will introduce inefficiencies with scheduling

Reply:
it's hard to say. handling a gossip message can do many things, and the message may be part of one of several protocols: tortoise beacon, hare, block, etc. what's more, the arrival of a gossip message can trigger a bunch of other things: database reads and writes, fetches of other data, etc. I don't think determining whether this is, say, CPU or IO bound, is that straightforward.

if we wanted to be really proper here, we'd use a different limit per protocol. but for now I think we can pick a reasonable number and research this further in future.

Reply:
let's follow up in #2708